File: tests/fixtures/sample.java
/**
 * Sample Java file for testing parser
 */

package com.example.test;

import java.util.ArrayList;
import java.util.List;

public class Calculator {
    private int result;

    public Calculator() {
        this.result = 0;
    }

    public int add(int a, int b) {
        this.result = a + b;
        return this.result;
    }

    public int subtract(int a, int b) {
        this.result = a - b;
        return this.result;
    }

    public static void main(String[] args) {
        Calculator calc = new Calculator();
        System.out.println("5 + 3 = " + calc.add(5, 3));
        System.out.println("10 - 4 = " + calc.subtract(10, 4));
    }
}

class Helper {
    public static void helloWorld() {
        System.out.println("Hello, World!");
    }
}

File: tests/fixtures/sample.ts
/**
 * Sample TypeScript file for testing parser
 */

interface Person {
  name: string;
  age: number;
}

function helloWorld(): void {
  console.log("Hello, World!");
}

class Calculator {
  private result: number;

  constructor() {
    this.result = 0;
  }

  add(a: number, b: number): number {
    this.result = a + b;
    return this.result;
  }

  subtract(a: number, b: number): number {
    this.result = a - b;
    return this.result;
  }
}

const greet = (name: string): string => {
  return `Hello, ${name}!`;
};

export { Calculator, greet, helloWorld, Person };
